compiler/rustc_errors/src/emitter.rs

```diff
@@ -2175,30 +2175,69 @@ impl EmitterWriter {
         file_lines: &FileLines,
         is_multiline: bool,
     ) {
-        // Print the span column to avoid confusion
-        buffer.puts(*row_num, 0, &self.maybe_anonymized(line_start + line_pos), Style::LineNumber);
         if let DisplaySuggestion::Diff = show_code_change {
             // Add the line number for both addition and removal to drive the point home.
             //
             // N - fn foo<A: T>(bar: A) {
             // N + fn foo(bar: impl T) {
+            let number_of_lines = file_lines.lines.len();
+            for (index, line_to_remove) in
+                file_lines.lines.iter().take(number_of_lines - 1).enumerate()
+            {
```

Comment:
Let's add a comment here explaining that we might be printing more than one line being removed. You can even mention https://github.com/rust-lang/rust/issues/92741 for full context for anyone trying to change this code in the future.